### PR TITLE
Adding autocrlf instructions for Windows system users

### PR DESCRIPTION
**Pull request by amoghrajesh**
During the contributors workshop this year, one user using windows ran into an issue where the windows machine was adding different line endings and breeze was behaving unexpectedly when commands were run. The fix to this was to add proper line endings for Git to use. Reference: https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by amoghrajesh** [comment]
@eladkal @potiuk adding this fix. Is this one note enough or do we need to add in more places?

**Comment by uranusjr** [comment]
I’d rather set this to false _for this repository only_. We can do this by intructing the user to set the config to *false* and without the `--global` flag. This ensures everyone on every platform uses Linux line endings for every file in this repository, and with modern Windows editors (including Notepad!) all being able to handle LF anyway, this should be good enough for any contributors.

Whether setting automatic CRLF conversion is a good idea varies from project to project, and I would most definitely prefer to not encourage a contirbutor to set this globally and pollute the entire machine. And in the case a project wants a non-default setting, *true* is almost never the right choice unless the project only targets Windows, which Airflow most certainly does not.

File: CONTRIBUTORS_QUICK_START.rst
Comment (hussein-awala) [inline CONTRIBUTORS_QUICK_START.rst]:
```suggestion
  $ git config core.autocrlf true
```

**Review by hussein-awala — APPROVED** [review]

**Comment by uranusjr** [comment]
Also, an alternative to instructing the user to configure this is to have `* text=auto` in our `.gitattributes` and ask the user to set `core.autocrlf=input` (again, only for this project, not globally). This would still allow the user to have CRLF in their working tree (not uncommon for some Windows editors to create new files in that line ending), but ensure the endings are converted on commit.

**Comment by amoghrajesh** [comment]
> I’d rather set this to false _for this repository only_. We can do this by intructing the user to set the config to _false_ and without the `--global` flag. This ensures everyone on every platform uses Linux line endings for every file in this repository, and with modern Windows editors (including Notepad!) all being able to handle LF anyway, this should be good enough for any contributors.
> 
> Whether setting automatic CRLF conversion is a good idea varies from project to project, and I would most definitely prefer to not encourage a contirbutor to set this globally and pollute the entire machine. And in the case a project wants a non-default setting, _true_ is almost never the right choice unless the project only targets Windows, which Airflow most certainly does not.

Good point. We should not do this globally at all. Coming to the value, does "false" seem like a more reasonable approach?
I would prefer special instructions for the windows user since they are less in number and having such a instruction would be an ideal scenario

**Comment by amoghrajesh** [comment]
@uranusjr I incorporated hussein's comment and set it locally to true for windows users. I do not know if this is the best solution, seems ideal. Let me know what you think here

**Comment by uranusjr** [comment]
There are two directions to the problem:

a. whether we want a Windows user’s _checkout_ to use CRLF, or just LF.
b. whether we want a file created by a Windows user with CRLF to be _checked in_ to the repository as-is, or automatically converted to use LF.

It used to be quite important to consider a. because many (older) Windows tools do not support the LF line ending well. This is however not a practical issue anymore; nowadays b. is the main consideration; since we (should) only want LF in the canonical repository on GitHub, a new file created on Windows must only be checked in with LF, not CRLF.

Now consider the possible configurations:

1. `core.autocrlf = false` would allow CRLF to be committed. There are other ways to prevent CRLF from being present though, so this is not entirely out of consideration. I think there’s a pre-commit hook for this, for example.
2. `core.autocrlf = true` would always convert line endings to CRLF on checkout, and convert line endings (back) to LF on checkin.
3. `core.autocrlf = input` is a combination of the two; it does not convert on checkout (so the local copy in the filesystem would use LF), but would convert to LF on checkin.

From a quick glance it’s easy to see option 2 and 3 seem more reasonable. The problem is however which files the conversions apply to. By default, Git would use a heuristic to figure out whether a file is text (and should be converted) or binary (and should not), but any heuristic is not perfect. Git therefore allows the repository to explicitly specify whether a file is text or not in `.gitattributes`, but still maintainers must be careful to review commits so no accidental checkins are made with wrongly detected file types. This is fortunately easy for Airflow specifically since we have a pretty solid review procedure, and GitHub’s interface makes the detection pretty easy.

So what is the best solution anyway? There isn’t one. In an ideal world I’d say it’s best to use `core.autocrlf = false`, use pre-commit to ensure no CRLF is committed in, and tell all contributors using Windows to be careful with their editor settings. But that’s not very practical for a public project like Airflow (more so for private projects, where you can just tell everyone on the team to set things up correctly), so it’s fine to use either `true` or `input` in practice and rely on maintainers to review and detect any errors Git’s algorithm may make. A pre-commit hook can still work in that case, although probably not be worthwhile since explaining what that lint error means to a novice contributor is probably harder than just telling the contributor to add and entry in `.gitattributes` anyway.

**Review by uranusjr — APPROVED** [review]

File: CONTRIBUTORS_QUICK_START.rst
Comment (uranusjr) [inline CONTRIBUTORS_QUICK_START.rst]:
```suggestion
Note: For windows based machines, on cloning, the Git line endings may be different from unix based systems
```

**Review by Lee-W — APPROVED** [review]

**Comment by amoghrajesh** [comment]
Thanks for the detailed response @uranusjr!
Seems ok to go with the current direction to unblock windows new users.

**Pull request merged**
